packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewerContainer.js

```diff
@@ -63,33 +63,43 @@ class LightCurveViewerContainer extends Component {
       } else {
         // Get the JSON data, or (as a failsafe) parse the JSON data if the
         // response is returned as a string
-        this.setState({ loading: asyncStates.success })
         return response.body || JSON.parse(response.text)
       }
     } catch (error) {
-      this.setState({
-        loading: asyncStates.error,
-        data: null
-      })
-      return error
+      return this.onError(error)
     }
   }
 
   async handleSubject () {
     try {
       const rawData = await this.requestData()
-      this.setState({
-        dataExtent: {
-          x: d3.extent(rawData.x),
-          y: d3.extent(rawData.y)
-        },
-        dataPoints: zip(rawData.x, rawData.y)
-      })
+      this.onLoad(rawData)
+      
     } catch (error) {
       console.error(error)
     }
   }
 
+  onLoad (rawData) {
+    this.setState({
+      dataExtent: {
+        x: d3.extent(rawData.x),
+        y: d3.extent(rawData.y)
+      },
+      dataPoints: zip(rawData.x, rawData.y),
+      loading: asyncStates.success
+    })
+    this.props.onReady()
```

Comment:
Perhaps this should be a callback of `setState`?

Reply:
Sure. I don't think it makes much difference here if `onReady` is called before, after, or at the same time as the state change.